drivers/SmartThings/zigbee-switch/src/aqara/init.lua

```diff
@@ -110,8 +111,12 @@ local function private_mode_handler(driver, device, value, zb_rx)
 end
 
 local function wireless_switch_handler(driver, device, value, zb_rx)
+  local ep_id = zb_rx.address_header.src_endpoint.value
+  if ep_id >= 0x29 then
+    ep_id = ep_id - 0x28
+  end
```

Comment:
`lumi.switch.b2laus01` uses endpoint as 0x29, 0x2A while `lumi.switch.n2acn1` uses 0x01, 0x02.

we have same code on the other part of driver like https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-switch/src/aqara/multi-switch/init.lua#L42-L45

Reply:
Yeah, I think it is only necessary to add the fingerprint here, since it's also in the sub-driver, that sub-driver's definition (which is identical) would be used.

Reply:
@greens Thank you for your feedback 😀 Is there anything specific I should change or improve in the code?

Reply:
I think all the changes except the addition of the fingerprint to this file can be removed.

Reply:
I don't agree. This code is necessary because the source address is different.
- Smart Wall Switch H1 Pro (With Neutral, Triple Rocker): 0x01, 0x02, 0x03
- Smart Wall Switch (No Neutral, Double Rocker): 0x29, 0x2A

Reply:
Right, but the `find_child` function set in the `multi_switch` sub-driver that you've added this device to should be called when `emit_event_for_endpoint` is here. This can be easily verified with the tests.

Reply:
Sorry, I didn't understand your comments. Can you tell me more specific? For example file path, name, line of code, etc. What do I need to fix? 🥲

Reply:
Revert the changes to `local function wireless_switch_handler(driver, device, value, zb_rx)` in this file.

Reply:
I can't agree with you. If the code reverted, the press-button event does not works.

```
2023-05-26T05:33:46.848374269+00:00 INFO Zigbee Switch  <ZigbeeDevice: c8ea856f-187f-4646-9ceb-5c39cb019595 [0x6EE5] (Aqara Smart Wall Switch (No Neutral, Double Rocker) 1)> received Zigbee message: < ZigbeeMessageRx || type: 0x00, < AddressHeader || src_addr: 0x6EE5, src_endpoint: 0x29, dest_addr: 0x0000, dest_endpoint: 0x01, profile: 0x0104, cluster: 0x0012 >, lqi: 0xCC, rssi: -50, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x18, seqno: 0x17, ZCLCommandId: 0x0A >, < ReportAttribute || < AttributeRecord || AttributeId: 0x0055, DataType: Uint16, Uint16: 0x0001 > > > >
2023-05-26T05:33:46.900857164+00:00 TRACE Zigbee Switch  Found ZigbeeMessageDispatcher handler in zigbee_switch -> Aqara Switch Handler
2023-05-26T05:33:46.904282876+00:00 INFO Zigbee Switch  Executing ZclClusterAttributeValueHandler: cluster: 0x0012, attribute: 0x0055
2023-05-26T05:33:46.908878173+00:00 ERROR Zigbee Switch  Aqara Smart Wall Switch (No Neutral, Double Rocker) 1 thread encountered error: [string "st/dispatcher.lua"]:233: Error encountered while processing event for <ZigbeeDevice: c8ea856f-187f-4646-9ceb-5c39cb019595 [0x6EE5] (Aqara Smart Wall Switch (No Neutral, Double Rocker) 1)>:
    arg1: < ZigbeeMessageRx || type: 0x00, < AddressHeader || src_addr: 0x6EE5, src_endpoint: 0x29, dest_addr: 0x0000, dest_endpoint: 0x01, profile: 0x0104, cluster: 0x0012 >, lqi: 0xCC, rssi: -50, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x18, seqno: 0x17, ZCLCommandId: 0x0A >, < ReportAttribute || < AttributeRecord || AttributeId: 0x0055, DataType: Uint16, Uint16: 0x0001 > > > >
[string "st/dispatcher.lua"]:233: Error encountered while processing event for <ZigbeeDevice: c8ea856f-187f-4646-9ceb-5c39cb019595 [0x6EE5] (Aqara Smart Wall Switch (No Neutral, Double Rocker) 1)>:
    arg1: < ZigbeeMessageRx || type: 0x00, < AddressHeader || src_addr: 0x6EE5, src_endpoint: 0x29, dest_addr: 0x0000, dest_endpoint: 0x01, profile: 0x0104, cluster: 0x0012 >, lqi: 0xCC, rssi: -50, body_length: 0x0008, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x18, seqno: 0x17, ZCLCommandId: 0x0A >, < ReportAttribute || < AttributeRecord || AttributeId: 0x0055, DataType: Uint16, Uint16: 0x0001 > > > >
[string "aqara/multi-switch/init.lua"]:44: attempt to compare number with nil
```

Reply:
that's because you forgot to also revert the line below to 

`device:emit_event_for_endpoint(zb_rx.address_header.src_endpoint.value,`

Reply:
Hi @greens ,
I look at the code of `aqara/multi-switch` driver. 

https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-switch/src/aqara/multi-switch/init.lua#L42-L45 <== This code seems redundant. (can be removed)

https://github.com/SmartThingsCommunity/SmartThingsEdgeDrivers/blob/main/drivers/SmartThings/zigbee-switch/src/aqara/multi-switch/init.lua#L54-L63
<== it sets `parent_assigned_child_key` as `2` not a `0x2a`. So I think there could be an issue when `device:emit_event_for_endpoint(0x29,...)`  . Could you let me know your opinion?

Reply:
You install your b2laus01 switch. Because the fingerprint is included in `multi-switch`, it uses that sub-driver's `init` method, and sets the `find_child` function for the device.

When `emit_event_for_endpoint` is called, it wraps a call to the device's `find_child` function. Because the `find_child` function is set for this device already to do the endpoint manipulation above, the event is emitted from the correct child device.

You can verify this for yourself by making the changes I described (revert all the changes made to this function) and running the tests locally.

Reply:
@greens , Ah.. Now I understand what you're mentioning. Thanks for the explaination